chain/chain/src/chain_update.rs

```diff
@@ -413,6 +419,7 @@ impl<'a> ChainUpdate<'a> {
         block: &Block,
         block_preprocess_info: BlockPreprocessInfo,
         apply_chunks_results: Vec<(ShardId, Result<ShardUpdateResult, Error>)>,
+        should_save_state_transition_data: bool,
```

Comment:
Can be computed here, because `postprocess_block` has `me`.

Reply:
That's true, but `Chain` also calls `ChainUpdate::process_apply_chunk_result`, which doesn't have a `me` argument. This means that `should_produce_state_witness_for_this_or_next_epoch` would have to be available from both `Chain` and `ChainUpdate`. I can't just do everything inside `ChainUpdate`.

I guess it would be possible to move  `should_produce_state_witness_for_this_or_next_epoch` to `EpochManager` and use it from both `Chain` and `ChainUpdate`, but I'm not convinced if there's any real benefit gained in exchange for this additional work. At this point I'm tempted to just merge it and unblock the statelessnet release.

Reply:
Uhh right. Okay, makes sense